docs/src/pages/demos/text-fields/ComposedTextField.js

```diff
@@ -31,23 +40,43 @@ class ComposedTextField extends React.Component {
     return (
       <div className={classes.container}>
         <FormControl className={classes.formControl}>
-          <InputLabel htmlFor="name-simple">Name</InputLabel>
-          <Input id="name-simple" value={this.state.name} onChange={this.handleChange} />
+          <InputLabel htmlFor="component-simple">Name</InputLabel>
+          <Input id="component-simple" value={this.state.name} onChange={this.handleChange} />
         </FormControl>
-        <FormControl className={classes.formControl} aria-describedby="name-helper-text">
-          <InputLabel htmlFor="name-helper">Name</InputLabel>
-          <Input id="name-helper" value={this.state.name} onChange={this.handleChange} />
-          <FormHelperText id="name-helper-text">Some important helper text</FormHelperText>
+        <FormControl className={classes.formControl} aria-describedby="component-helper-text">
+          <InputLabel htmlFor="component-helper">Name</InputLabel>
+          <Input id="component-helper" value={this.state.name} onChange={this.handleChange} />
+          <FormHelperText id="component-helper-text">Some important helper text</FormHelperText>
         </FormControl>
         <FormControl className={classes.formControl} disabled>
-          <InputLabel htmlFor="name-disabled">Name</InputLabel>
-          <Input id="name-disabled" value={this.state.name} onChange={this.handleChange} />
+          <InputLabel htmlFor="component-disabled">Name</InputLabel>
+          <Input id="component-disabled" value={this.state.name} onChange={this.handleChange} />
           <FormHelperText>Disabled</FormHelperText>
         </FormControl>
-        <FormControl className={classes.formControl} error aria-describedby="name-error-text">
-          <InputLabel htmlFor="name-error">Name</InputLabel>
-          <Input id="name-error" value={this.state.name} onChange={this.handleChange} />
-          <FormHelperText id="name-error-text">Error</FormHelperText>
+        <FormControl className={classes.formControl} error aria-describedby="component-error-text">
+          <InputLabel htmlFor="component-error">Name</InputLabel>
+          <Input id="component-error" value={this.state.name} onChange={this.handleChange} />
+          <FormHelperText id="component-error-text">Error</FormHelperText>
+        </FormControl>
+        <FormControl className={classes.formControl} variant="outlined">
+          <InputLabel
+            ref={ref => {
+              this.labelRef = ReactDOM.findDOMNode(ref);
```

Comment:
I believe you don't need to do this anymore. The ref already grants access to the dom node.

Reply:
How?

Reply:
Well not in this case because InputLabel is not properly forwarding the ref. I assumed it did which was wrong. I always try to avoid `findDOMNode` since it's usage in refs is [discourage by React](https://reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components).

If it would be forwarded we could reduce the boilerplate you wanted to avoid by using callback refs by simply using `createRef` and then passing that to `InputLabel#ref`.

facebook/react#4936 has additional explanation about `findDOMNode(ref)`.

Considering how much customization we allow `findDOMNode` is probably safer and since it works there's no need for a change in my book. I just spotted something unusual (for me).

Reply:
As far as I understand the warning about using `findDOMNode`, it's about educating people that React might provide another way to do what they want. Many people access the DOM when they don't need to. Hence the warning. In our case, it's fine.